src/infrastructure/design/singleton.ts

```diff
@@ -0,0 +1,26 @@
+import { AppError } from 'ng2-qgrid/core/infrastructure';
+
+function decorate(method) {
+	let instance;
+	let hasInstance = false;
+	return function value() {
```

Comment:
use arrow function?

Reply:
i can't do it, cause this is used inside)